internal/scheduler/configuration/configuration.go

```diff
@@ -73,6 +73,13 @@ type MetricsConfig struct {
 	TrackedErrorRegexes []string
 	// Metrics are exported for these resources.
 	TrackedResourceNames []v1.ResourceName
+	// Optionally rename resources in exported metrics.
+	// E.g., if ResourceRenaming["nvidia.com/gpu"] = "gpu", then metrics for resource "nvidia.com/gpu" use resource name "gpu" instead.
+	// This can be used to avoid illegal Prometheus metric names (e.g., for "nvidia.com/gpu" as "/" is not allowed).
+	// Allowed characters in resource names are [a-zA-Z_:][a-zA-Z0-9_:]*
+	// If a tracked resource contains illegal characters, it is should be renamed here.
+	// It can also be used to track multiple resources within the same metric, e.g., "nvidia.com/gpu" and "amd.com/gpu".
+	ResourceRenaming map[v1.ResourceName]string
```

Comment:
We're using the validator package for config validation elsewhere. https://pkg.go.dev/github.com/go-playground/validator/v10#section-readme
Could you use this pre-defined regex to validate that these metric names are valid? https://github.com/go-playground/validator/blob/v10.19.0/regexes.go#L7

Reply:
this doesn't include underscores or colons https://github.com/go-playground/validator/blob/a0f74b0fb2a7ae1750c0f0b0a49550d8b6e2e708/regexes.go#L7

Also, it allows a metric that starts with a digit, which prometheus will reject later.  

most tracked resources doesn't include underscores or colons themselves, however this might reject valid resource names in the future. 

The regex i added acts the same as prometheus naming check (in the current version we are using ) https://github.com/prometheus/common/blob/49b36038aead362ce78f09337dc5872000bd048a/model/metric.go#L97